tensorrt_llm/scaffolding/contrib/DeepThink/deep_think_controller.py

```
@@ -0,0 +1,79 @@
+        confidence_info.conf_grouped += new_conf
+
+
+class DeepThinkOfflineController(NativeGenerationController):
+
+    def __init__(self, conf_group_size: int, conf_threshold: float, sampling_params: dict = None, streaming: bool = False):
+        super().__init__(sampling_params, streaming)
+        self.confidence_info = ConfidenceInfo(conf_grouped=0.0, conf_list=[], conf_group_list=deque([]), conf_group_size=conf_group_size, conf_threshold=conf_threshold)
+
+    def process(self, tasks: List[Task], **kwargs):
+        yield from super().process(tasks, **kwargs)
+        assert (len(tasks) == 1, "DeepThinkOfflineController only supports one task")
```

Ruff (0.12.2)

Assert test is a non-empty tuple, which is always `True`

(F631)